docs/generatestepdocs.py

```diff
@@ -0,0 +1,203 @@
+import sys
+import os
+import collections
+
+#used to switch to various predefined ways the final table can look
+"""
+Default 'file' shows 
+---
+annotation | file name | line number
+signature
+---
+compact
+---
+annotation
+signature
+---
+ansig
+---
+annotation | signature 
+---
+"""
+STYLE = "compact"
+
+OUTPUTDIR = "docs/"
+
+steps = []
+
+all_steps = []
+
+Annotation = collections.namedtuple("Annotation", ["filename", "start", "end", "regex", "signature"])
+
+def setup_filter():
+    global annotation_filter
+    if len(sys.argv) == 3:
+        f = sys.argv[2]
+        f = f.split(",")
+        for a in f:
+            annotation_filter.add(a)
+    else: 
+        annotation_filter = set(["when", "then", "given", "and", "but"])
+
+def print_help():
+    print("""
+        USAGE: generatestepdocs.py <directories to generate docs from> [filter] \n
```

Comment:
change the name of the file and this usage, please, for example to camelCase (to be more readable)